src/main.rs

```diff
@@ -103,6 +105,18 @@ fn get_gps(config: Rc<Config>) -> Box<GPS> {
 
                 _ => println!("Failed to open serial device: {}", e),
             }
+        }
+    }
+
+    match GNSS::new(config.clone()) {
```

Comment:
Wouldn't having this seperate/independent from the RS232 above, mean we'll **always** get the `Failed to open serial device` from above match block for GNSS devices? Perhaps i'd be better to do GNSS detection if the RS232 code fails and error is not `NotFound`?

Reply:
This is my first attempts at rust so I think i understand how I might do that. I'll update the PR

Reply:
I updated the code to check all devices before printing the error.